docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,294 @@
+
+        import weaviate
+
+        if config.client is None:
+            self._client = weaviate.Client('http://localhost:8080')
+        elif isinstance(config.client, str):
+            self._client = weaviate.Client(config.client)
+        else:
+            self._client = config.client
+
+        if config.name is not None and docs is not None:
+            raise ValueError(
+                'only one of name or docs can be provided for initialization'
+            )
+
+        self._schemas = self._load_or_create_weaviate_schema(config.name)
+        self._offset2ids, self._offset2ids_wid = self._get_offset2ids_meta()
+
+        if docs is None or config.name:
+            return
+
+        elif isinstance(
+            docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
+        ):
+            self.extend(Document(d, copy=True) for d in docs)
+        else:
+            if isinstance(docs, Document):
+                self.append(docs)
+
+    def _get_weaviate_class_name(self) -> str:
+        """Generate the class/schema name using the ``uuid1`` module with some
+        formatting to tailor to weaviate class name convention
+
+        :return: string representing the name of  weaviate class/schema name of
+            this :class:`DocumentArrayWeaviate` object
+        """
+        return ''.join([i for i in uuid.uuid1().hex if not i.isdigit()]).capitalize()
+
+    def _get_schema_by_name(self, cls_name: str) -> Dict:
+        """Return the schema dictionary object with the class name
+        Content of the all dictionaries by this method are the same except the name
+        of the weaviate's ``class``
+
+        :param cls_name: the name of the schema/class in weaviate
+        :return: the schema dictionary
+        """
+        # TODO: ideally we should only use one schema. this will allow us to deal with
+        # consistency better
+        return {
+            'classes': [
+                {
+                    'class': cls_name,
+                    "vectorizer": "none",
+                    # TODO: this skips checking embedding dimension but might not
+                    # work if want to leverage weaviate for vector search
+                    'vectorIndexConfig': {'skip': True},
+                    'properties': [
+                        {
+                            'dataType': ['blob'],
+                            'name': '_serialized',
+                            'indexInverted': False,
+                        },
+                    ],
+                },
+                {
+                    'class': cls_name + 'Meta',
+                    "vectorizer": "none",
+                    'vectorIndexConfig': {'skip': True},
+                    'properties': [
+                        {
+                            'dataType': ['string[]'],
+                            'name': '_offset2ids',
+                            'indexInverted': False,
+                        },
+                    ],
```

Comment:
Does it make sense to create a property of `string` instead where the whole column consists a mapping between offsets to the document IDs. Then, we no more need to load offset2ids in memory and just need to keep it in the database.
We just need to convert offsets to uuids when we want to do the mapping (maybe something like `uuid.uuid5(uuid.NAMESPACE_URL, str(offset))`)

Reply:
Do you mean creating a new field of type `string` in the `Document` schema (i.e. schema with name `_class_name`, not `_meta_name`) that maps offset -> document ID? If so, yes combining the two schemas into one is the plan going forward but will be tackled in next phase (as detailed in google docs/description). 

Currently, what blocks us from doing that is that unlike SQLite where we can update the entire column with a single SQL statement, in Weaviate's case I don't think it's possible to update an entire column (or batch update every property of objects in a class) atm (their Python client only allows adding objects in batch, not updating/deleting). 

If I do not understand you correctly, would you please provide an example (e.g. `daw.insert(3, doc)` or `del daw[4]`)  to illustrate what would happen in your suggested case? Thank you.

Reply:
Basicly I meant something like:
```
                {
                    'class': cls_name + 'Meta',
                    "vectorizer": "none",
                    'vectorIndexConfig': {'skip': True},
                    'properties': [
                        {
                            'dataType': ['string[]'],
                            'name': '_offset2ids',
                            'indexInverted': False,
                        },
                    ]
                }
```
and implementing `def _offset2id(self, offset)` so that it converts offset to a weaviate id first (like in wmap) and then retrieves to id associated with offset.
But now I get your point, weaviate doesn't support batch operations and this will be problematic in some cases.